client/httplib/httplib_test.go

```diff
@@ -345,6 +345,28 @@ func TestNewBeegoRequest(t *testing.T) {
 	req := NewBeegoRequest("http://beego.me", "GET")
 	assert.NotNil(t, req)
 	assert.Equal(t, "GET", req.req.Method)
+
+	req = NewBeegoRequest("httpa\ta://beego.me", "GET")
```

Comment:
A typo I guess.

```suggestion
	req = NewBeegoRequest("http://beego.me", "GET")
```

Reply:
This is a invalid url test case. 
The NewRequest alway return not nil result and I believe it's not a good behavior. 
![image](https://user-images.githubusercontent.com/9923838/105628276-fedd4d00-5e76-11eb-893f-5c9cde3d6f37.png)
We should return (*BeegoHTTPRequest, error).

But this is a public API, so I am not sure when to do this change.

Reply:
> This is a invalid url test case.

Oh, I apologise, I didn't get this right away.